pkg/api/v1beta3/types.go

```diff
@@ -866,6 +866,53 @@ type PodTemplateList struct {
 	Items []PodTemplate `json:"items" description:"list of pod templates"`
 }
 
+// JobSpec is the specification of a job.
+type JobSpec struct {
+	// Completions is the number of desired completions.
+	Completions int `json:"completions" description:"number of times the job has completed successfully (desired or observed, as appropriate)"`
+
+	// Selector is a label query over pods that should match the Completions count.
+	// If Selector is empty, it is defaulted to the labels present on the Pod template.
+	Selector map[string]string `json:"selector,omitempty" description:"label keys and values that must match in order to be controlled by this job, if empty defaulted to labels on Pod template"`
```

Comment:
As with RC, we should default the selector and labels to the pod template's labels.